core/string_api.php

```diff
@@ -237,6 +237,15 @@ function string_url( $p_string ) {
 	return rawurlencode( $p_string );
 }
 
+/**
+ * Build a URL-encoded query string from an array of key value pairs.
+ * @param  array $p_params Query string parameters.
+ * @return string
+ */
+function string_build_query( array $p_params ): string {
```

Comment:
For consistency, shouldn't this new API function replace other, existing usages of http_build_query() ?

Otherwise we'll end up with some URLs RFC1738-encoded (default for $encoding_type parameter). Maybe not an issue, did not check.

Reply:
That's unfortunately something that must be expected with a 25-year-old code base 😉

Reply:
I think that if this PR is accepted, a separate issue should be opened for string_build_query() using.

Reply:
Opened https://mantisbt.org/bugs/view.php?id=35424 to track this.